ethcore/src/miner/transaction_queue.rs

```diff
@@ -690,18 +707,61 @@ impl TransactionQueue {
 		G: Fn(&SignedTransaction) -> U256,
 	{
 
-		if tx.gas_price < self.minimal_gas_price && origin != TransactionOrigin::Local {
-			trace!(target: "txqueue",
-				"Dropping transaction below minimal gas price threshold: {:?} (gp: {} < {})",
-				tx.hash(),
-				tx.gas_price,
-				self.minimal_gas_price
-			);
+		if origin != TransactionOrigin::Local && tx.gas_price < self.minimal_gas_price {
```

Comment:
As I understand from the code, local transactions were always accepted, even if gas_price was too low => added zgp-checks for non-local transactions only.